contracts/utils/convex/ConvexStakingWrapperYield.sol

```diff
@@ -0,0 +1,111 @@
+
+    constructor() public {}
+
+    function initialize(
+        address _curveToken,
+        address _convexToken,
+        address _convexPool,
+        uint256 _poolId,
+        address _vault
+    ) external override auth{
+        require(!isInit, 'already init');
+        owner = address(0xa3C5A1e09150B75ff251c1a7815A07182c3de2FB); //TODO: Find why this needs to be set to convex multisig
+        emit OwnershipTransferred(address(0), owner);//TODO: Find why this needs to be done
+        _tokenname = string(abi.encodePacked('Staked ', ERC20(_convexToken).name(), ' Yield'));
+        _tokensymbol = string(abi.encodePacked('stk', ERC20(_convexToken).symbol(), '-yield'));
+        isShutdown = false;
+        isInit = true;
+        curveToken = _curveToken;
+        convexToken = _convexToken;
+        convexPool = _convexPool;
+        convexPoolId = _poolId;
+        collateralVault = _vault; //TODO: Add the join address
+
+        //add rewards
+        addRewards();
+        setApprovals();
+    }
+
+    function setCauldron(address _cauldron) external auth{
+        require(_cauldron!=address(0), 'cauldron address cannot be 0');
+        cauldron = _cauldron;
+    }
+
+    // Set the locations of vaults where the user's funds have been deposited & the accounting is kept
+    function setVault(address _account, bytes12 _vault) external auth {
+        bytes12[] storage userVault = vaults[_account];
+        for (uint256 i = 0; i < userVault.length; i++) {
+            require(userVault[i] != _vault, 'already added');
+        }
+        userVault.push(_vault);
+        vaults[_account] = userVault;
+    }
+
+    function removeVault(address _account, bytes12 _vault) external auth {
+        bytes12[] storage userVault = vaults[_account];
+        for (uint256 i = 0; i < userVault.length; i++) {
+            if(userVault[i] == _vault){
+                vaults[_account] = remove(i,userVault);
+                break;
+            }
+        }
+    }
+
+    function remove(uint _index,bytes12[] storage userVault) internal returns (bytes12[] memory){
+        require(_index < userVault.length, "index out of bound");
+
+        for (uint i = _index; i < userVault.length - 1; i++) {
+            userVault[i] = userVault[i + 1];
+        }
+        userVault.pop();
+        return userVault;
+    }
+
+    // Get user's balance of collateral deposited at in various vaults
+    function _getDepositedBalance(address _account) internal view override returns (uint256) {
+        if (_account == address(0) || _account == collateralVault) {
+            return 0;
+        }
+
+        if (vaults[_account].length == 0) {
+            return balanceOf(_account);
+        }
+    bytes12[] memory userVault = vaults[_account];
+        //add up all balances of all vaults
+        uint256 collateral;
+        for (uint256 i = 0; i < userVault.length; i++) {
+            try ICauldron(cauldron).balances(userVault[i]) returns (Balances memory _balance) {
```

Comment:
No need to use `try... catch`, if you read `cauldron.balances` on a non existing `vaultId` you will get an empty `Balances` struct, with `ink` and `art` set to zero.

So you can just assume the balances exist and add them.